storm-shim/src/main/java/storm/mesos/shims/ICommandLineShim.java

```diff
@@ -0,0 +1,12 @@
+package storm.mesos.shims;
+
+/**
+ * Returns the command line, shimmed and 0.9.X and 0.10.X have slightly a different command
+ */
+public interface ICommandLineShim {
+
+  public String getCommandLine();
+
+  public String getDockerCommandLine();
```

Comment:
The Factory class can certainly live in the storm-mesos module so there's really no more maven work.  Aesthetically it may be more pleasing.  The one negative is there's slightly more code for each new shim (storm 0.11.0?) with @choang's approach, while the Mesos-Nimbus code stays constant with either approach.  Are there strong opinions here?